bundle/regal/ast.rego

```diff
@@ -269,6 +269,18 @@ all_refs := [value.value |
 	value.type == "ref"
 ]
 
+ref_to_string(ref) := concat(".", name_parts) if {
+	name_parts := array.concat([ref[0].value], [s |
+		some i, part in ref
+		i > 0
+		s := _ref_part_to_string(part)
+	])
+}
+
+_ref_part_to_string(ref) := ref.value if ref.type == "string"
+
+_ref_part_to_string(ref) := concat("", ["$", ref.value]) if ref.type != "string"
```

Comment:
[style] I think I'd cram more into the helper 🤔 

```suggestion
ref_to_string(ref) := concat(".", [ _ref_part_to_string(i, part) | some i, part in ref ])

_ref_part_to_string(0, ref) := ref.value
_ref_part_to_string(_, ref) := ref.value if ref.type == "string"
_ref_part_to_string(_, ref) := concat("", ["$", ref.value]) if {
    ref.type != "string"
    i > 0
}
```

Reply:
Oh, that's nice! I'll go with that 👍